nixos/modules/virtualisation/azure-agent.nix

```diff
@@ -23,21 +24,23 @@ let
                     procps # for pidof
                     shadow # for useradd, usermod
                     utillinux # for (u)mount, fdisk, sfdisk, mkswap
-                    parted
+                    parted bash
                   ];
     pythonPath = [ pythonPackages.pyasn1 ];
 
     configurePhase = false;
     buildPhase = false;
 
     installPhase = ''
+      substituteInPlace config/66-azure-storage.rules \
+          --replace /bin/sh "${bash}/bin/sh"
       substituteInPlace config/99-azure-product-uuid.rules \
           --replace /bin/chmod "${coreutils}/bin/chmod"
       mkdir -p $out/lib/udev/rules.d
       cp config/*.rules $out/lib/udev/rules.d
 
       mkdir -p $out/bin
-      cp waagent $out/bin/
+      cp bin/waagent2.0 $out/bin/waagent
```

Comment:
Why is this now named differently upstream? Doesn't this usually indicate backwards incompatibility?

Reply:
Yes this is to point out this is a new version of the agent. The old main remains only for backward compatibility, but is just calling the new one [waagent](https://github.com/Azure/WALinuxAgent/blob/v2.2.36/bin/waagent).

Reply:
But shouldn't we also call it `waagent2.0` then? If upstream names it that we should do so too, unless I'm missing something

Reply:
```suggestion
      cp bin/waagent2.0 $out/bin/waagent2.0
```